File: Framework/Qt/Python/Chapter_01/QT01_06.py
"""
Object-oriented programming of QT_05 example.
"""

from PySide2.QtWidgets import QApplication, QMainWindow, QWidget, QLabel, QVBoxLayout

class MainWindow(QMainWindow):
    def __init__(self):
        # INHERIT attributes from superclass QMainWindow.
        super().__init__()

        widget = QWidget()
        layout = QVBoxLayout(widget)

        label1 = QLabel("Hello World!")
        label2 = QLabel("Goodbye World?")

        layout.addWidget(label1)
        layout.addWidget(label2)

        # SET empty widget as the Central Widget of the current object (denoted "self").
        self.setCentralWidget(widget)
        self.show()


if __name__ == '__main__':
    app = QApplication()

    # DECLARE Qt application software's main window via MainWindow.
    window = MainWindow()

    app.exec_()

File: Framework/Qt/Cpp/Chapter_01/QT01_06/QT01_06.cpp
/*
Setup Qt application software using empty widget containing child widgets.
*/

#include <qapplication.h>
#include "MainWindow.h"

int main(int argc, char **argv) {
	QApplication app(argc, argv);

	// DECLARE Qt application software's main window via MainWindow.
	MainWindow window;

	return app.exec();
}